src/os.rs

```diff
@@ -417,6 +418,36 @@ mod imp {
 }
 
 
+#[cfg(feature = "rdrand")]
+mod imp {
+    #[cfg(not(feature="no_std"))] use std::io;
+    #[cfg(feature="no_std")] use io;
+
+    use Rng;
+
+    pub struct OsRng;
+
+    impl OsRng {
+        pub fn new() -> io::Result<OsRng> {
+            Ok(OsRng)
+        }
+    }
+
+    impl Rng for OsRng {
+        fn next_u32(&mut self) -> u32 {
+            let ret;
+            unsafe{asm!("rdrand $0":"=r"(ret):::"volatile")};
```

Comment:
You didn't check the carry bit to ensure that RDRAND succeeded. RDRAND can and does fail.